src/Umbraco.Web.UI.Client/src/common/directives/components/umbrangeslider.directive.js

```diff
@@ -299,7 +300,16 @@ For extra details about options and events take a look here: https://refreshless
                 });
             });
         }
-
+      function setUpActivePipsHandling() {
+        let activePip = [null, null];
+        sliderInstance.noUiSlider.on('update', function (values,handle) {
+          if(activePip[handle]){
+            activePip[handle].classList.remove("noUi-value-active");
+          }
+          activePip[handle] = sliderInstance.querySelector('.noUi-value[data-value="' + parseInt(values[handle]) + '"]');
```

Comment:
`parseInt()` makes sense in the telemetry slider, but it other sliders the values may be decimals.

So it should probably use this approach instead: https://refreshless.com/nouislider/examples/#section-pips

```
// Match the formatting for the pip
var dataValue = Math.round(values[handle]);

// Find the pip matching the value
activePips[handle] = pipsSlider.querySelector('.noUi-value[data-value="' + dataValue + '"]');
```

Reply:
ParseInt should work as long as all the decimal places are zero. Do you see a scenario where it is something else?

Reply:
I get it now, I'll look into it, on nouislider the value can also be a string if I recall correctly. But I'm not sure the component allows it, I'll test some things

Reply:
I've played around with it for a bit and Math.round won't work either.

So if the data-value attribute is `2.1` the update events values array will have it written as `2.10` which makes it hard to query using the attribute eg. something like this would **not** work with decimals `querySelector('.noUi-value[data-value="' + dataValue + '"]');`

Right now I'm looping through the pips and checking each data-value on update, which is not ideal. However, it works with numbers and decimals.

Reply:
The `<umb-range-slider>` could be used in a custom property editor, dashboard etc. where the pips could be decimal steps. E.g. a step size on 0.1

Not sure how the pips are displayed and if only 1, 2, 3 etc would be marked with the "active" class.

Reply:
With the new implementation, the active class will work fine for pips with decimals. Though it is a little inefficient compared to before